napalm/eos/eos.py

```diff
@@ -131,60 +136,120 @@ def _process_optional_args(self, optional_args):
-            # does not raise an Exception if unusable
+                if self.device is None:
+                    self.device = Node(connection, enablepwd=self.enablepwd)
+                # does not raise an Exception if unusable
 
-            # let's try to determine if we need to use new EOS cli syntax
-            sh_ver = self.device.run_commands(["show version"])
-            cli_version = (
-                2 if EOSVersion(sh_ver[0]["version"]) >= EOSVersion("4.23.0") else 1
-            )
+                # let's try to determine if we need to use new EOS cli syntax
+                sh_ver = self.device.run_commands(["show version"])
+                self.cli_version = (
+                    2 if EOSVersion(sh_ver[0]["version"]) >= EOSVersion("4.23.0") else 1
+                )
 
-            self.device.update_cli_version(cli_version)
-        except ConnectionError as ce:
-            # and this is raised either if device not avaiable
-            # either if HTTP(S) agent is not enabled
-            # show management api http-commands
-            raise ConnectionException(str(ce))
+                self.device.update_cli_version(self.cli_version)
+            except ConnectionError as ce:
+                # and this is raised either if device not avaiable
+                # either if HTTP(S) agent is not enabled
+                # show management api http-commands
+                raise ConnectionException(str(ce))
 
     def close(self):
         """Implementation of NAPALM method close."""
         self.discard_config()
 
     def is_alive(self):
+        if self.transport == "ssh":
+            null = chr(0)
+            if self.device is None:
+                return {"is_alive": False}
+            try:
+                # Try sending ASCII null byte to maintain the connection alive
+                self.device.write_channel(null)
+                return {"is_alive": self.device.remote_conn.transport.is_active()}
+            except (socket.error, EOFError):
+                # If unable to send, we can tell for sure that the connection is unusable
+                return {"is_alive": False}
+
         return {"is_alive": True}  # always true as eAPI is HTTP-based
 
+    def _run_commands(self, commands, **kwargs):
+        if self.transport == "ssh":
+            if self.fn0039_transform:
+                if isinstance(commands, str):
+                    commands = [cli_convert(commands, self.cli_version)]
+                else:
+                    commands = [cli_convert(cmd, self.cli_version) for cmd in commands]
+            ret = []
+            for command in commands:
+                if kwargs.get("encoding") == "text":
+                    cmd_output = self._netmiko_device.send_command(command).replace(
+                        "% Invalid input", ""
+                    )
+                    ret.append({"output": cmd_output})
+                    continue
+
+                cmd_pipe = command + " | json"
```

Comment:
This might have to be ` | json version 2`.  I can't remember exactly which version/reversion of JSON output EAPI uses.  Might have to ask Arista for exact clarification.

Reply:
CC @thomasbridge74 - I let you investigate this, feel free to come up with a subsequent PR if needed.